lib/dialects/mssql/query-generator.js

```diff
@@ -465,7 +465,7 @@ class MSSQLQueryGenerator extends AbstractQueryGenerator {
        * Exclude NULL Composite PK/UK. Partial Composite clauses should also be excluded as it doesn't guarantee a single row
        */
       for (const key in clause) {
-        if (typeof clause[key] === 'undefined' || clause[key] == null) {
+        if (primaryKeysAttrs.includes(key) && (typeof clause[key] === 'undefined' || clause[key] === null)) {
```

Comment:
Can you also do the following refactor please, since you're already editing this line?

```suggestion
        if (primaryKeysAttrs.includes(key) && (clause[key] === undefined || clause[key] === null)) {
```